src/zeek-setup.cc

```diff
@@ -535,7 +535,11 @@ SetupResult setup(int argc, char** argv, Options* zopts)
 		{
 		// The supervisor only needs to load a limited set of
 		// scripts, since it won't be doing traffic processing.
-		if ( options.supervisor_mode )
+		// The same goes for nodes it creates that aren't part
+		// of the data cluster -- those nodes establish an
+		// environment variable ZEEK_CLUSTER_MGMT_NODE that defines
+		// their role (so far: controller or agent).
+		if ( options.supervisor_mode || getenv("ZEEK_CLUSTER_MGMT_NODE") )
```

Comment:
Yeah, I'm not really happy with that part either. The idea here is that the supervisor-level agent/controller code tell the created node what role it serves. Similarly to the question of what the supervisor itself runs, I didn't want the forked node to load all scripts — I figured the agent process just need to load the agent scripts, etc.

When one tells the supervisor to create a node, one can tell it to load specific scripts _in addition_ to what it would normally load (which would be `init-default`). I considered using that, but it doesn't give me a way to say what _not_ to load.

One thought would be to convey the init script itself via the environment. I.e., if you set `ZEEK_INIT_SCRIPT` to `init-default`, Zeek will load that, if you say `init-supervisor`, it'll be that, etc. It'd keep the cluster management out of it.